components/login/SignUpForm.js

```diff
@@ -1,77 +1,81 @@
+  };
 
-    const handleEmail = (newValue) => {
-        setSignupEmail(newValue);
-    };
+  const handleSetPassword = (newValue) => {
+    setSignupPassword(newValue);
+  };
 
-    const handlePassword = (newValue) => {
-        setSignupPassword(newValue);
-    };
+  const handleSetPassConf = (newValue) => {
+    setSignupPassConf(newValue);
+  };
 
-    const handlePassConf = (newValue) => {
-        setSignupPassConf(newValue);
-    };
-
-    const handleSignup = (e) => {
-        e.preventDefault();
-        if (signupPassword !== signupPassConf) {
-            setSignupNotification(
-                'Password and password confirmation do not match'
-            )
-            setSignupPassword('');
-            setSignupPassConf('');
-            return null;
-        }
-        createUserWithEmailAndPassword(auth, signupEmail, signupPassword)
-            .catch((err) => {
-                console.log(err.code, err.message)
-                setSignupNotification(err.message)
-            });
+  const handleSignUp = (e) => {
+    e.preventDefault();
+    if (signupPassword !== signupPassConf) {
+      setSignupNotification("Password and password confirmation do not match!");
+      setSignupPassword("");
+      setSignupPassConf("");
+      return null;
     }
+    createUserWithEmailAndPassword(auth, signupEmail, signupPassword)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        console.log(`The user '${user.uid}' has signed up.`);
+        router.push({
+          pathname: "/[user]",
+          query: {
+            user: user.uid,
+            name: signupEmail.match(/^([^@]*)@/),
+          },
+        });
+      })
+      .catch((err) => {
+        console.log(err.code, err.message);
+        setSignupNotification(err.message);
+      });
+    setSignupEmail("");
+    setSignupPassword("");
+    setSignupPassConf("");
+  };
 
-    return (
-        <div>
-            <h1>Sign up</h1>
-            <form onSubmit={handleSignup}>
-                <EmailInputWithLabel
-                    value={signupEmail}
-                    onChange={handleEmail}
-                />
-                <br />
-                <PasswordInputWithLabel
-                    value={signupPassword}
-                    onChange={handlePassword}
-                    labelTextContent='Password'
-                />
-                <br />
-                <PasswordInputWithLabel
-                    value={signupPassConf}
-                    onChange={handlePassConf}
-                    labelTextContent='Confirm password'
-                />
-                <br />
-                <SubmitButton buttonTextContent='Sign up' />
-                <br />
-                {signupNotification}
-            </form>
-        </div>
-    );
+  return (
+    <form onSubmit={handleSignUp}>
+      <h2>Sign up</h2>
+      <EmailInputWithLabel value={signupEmail} onSetEmail={handleSetEmail}>
+        &#128231;
```

Comment:
I think that labels content is essential for the accessibility machine-reading and replacing words with emojis is not the best practice for such a purpose. In order to have a neat and less space-consuming input fields appearance, I would suggest having visible text placeholders inside the input fields describing their purpose (hiding on focusing) instead of labels while having the labels styled invisible just for machine-reading. In this case, having an invisible emoji would have less sense than plain text as label content.

Reply:
Are you worried about the app even exist or how you can implement a minuscule bit of some accessibility concern evenwhen you don't have any version of an app working prototype? This pattern of putting-the-cart-before-the-horse has been discussed almost at every brief since day 1 of the project. What's the point to discuss a CSS issue when you don't even have an app to even bother thnking about hwo you style it? And as I said code your stuff before you mouth it. And this accessibility concern should always come after we decide how the elements would be styled and with what. But you started to insist on it when 1) you didn't show how it work and just threw a CSS class inside of an element without the primary content suggested in it, 2) _we didn't even have an app to begin looking at CSS_. In fact, React children prop doesn't have to be text and can be anything, including a whole chunk of html elements with whole lots of CSS classes assigned to it, so it's not even an issue to include accessibility implementations inside the children prop if we want to. You're raising an issue of no issue that only yourself don't know that's no issue and insist on what you believe was a eureka that's not.